psi4/src/psi4/libmints/basisset.cc

```diff
@@ -859,13 +805,13 @@ BasisSet::BasisSet(const std::string &basistype, SharedMolecule mol,
                 shells_[shell_count] =
                     GaussianShell(shelltype, am, shell_nprim, &uoriginal_coefficients_[ustart + atom_nprim],
                                   &ucoefficients_[ustart + atom_nprim], &uerd_coefficients_[ustart + atom_nprim],
-                                  &uexponents_[ustart + atom_nprim], puream, n, xyz_ptr, bf_count);
+                                  &uexponents_[ustart + atom_nprim], puream, n, &xyz_.data()[3 * n], bf_count);
```

Comment:
Also this is confusing. Previously, we passed in the pointer (as opposed to a reference), which if I'm reading this right would point to 0 for n = 0, and then due to the `xyz_ptr += 3`, it works out to point in the same field as [3*n]. Correct?